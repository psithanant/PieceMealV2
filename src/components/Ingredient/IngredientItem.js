import React, { Component } from 'react';
import { TouchableOpacity } from 'react-native';
import {
    Icon,
    Card,
    CardItem,
    Thumbnail,
    Text,
    Right } from 'native-base';
import DEFAULT_IMAGE from '../../assets/food/ice-cream.jpg';

export default class IngredientItem extends Component {
  render() {
    const rowData = this.props.rowData;
    let name = rowData.name || '';
    name = name.split(' ').map((str) => {
      return str.substring(0, 1).toUpperCase() + str.substring(1);
    }).join(' ');
    let image;
    if (rowData.image_url) {
      image = { uri: rowData.image_url };
    } else {
      image = DEFAULT_IMAGE;
    }
    return (
        <Card>
          <TouchableOpacity onPress={this.props.onPress}>
            <CardItem>
              <Thumbnail source={image} />
                <Text style={styles}>{name}</Text>
                <Right>
                  <Icon name="arrow-forward" style={{ color: '#68BAA7' }} />
                </Right>
            </CardItem>
          </TouchableOpacity>
        </Card>
    );
  }
}

const styles = {
  alignSelf: 'center',
  marginLeft: 25,
  fontFamily: 'Futura',
  fontSize: 16,
  color: '#373737'
};
