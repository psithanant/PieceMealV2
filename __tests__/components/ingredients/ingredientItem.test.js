import 'react-native';
import React from 'react';
import IngredientItem from '../../../src/components/Ingredient/IngredientItem';
import renderer from 'react-test-renderer';

describe('tests the IngredientItem component', () => {
  it('renders IngredientItem component properly', () => {
    expect(renderer.create(
      <IngredientItem
        rowData={{id: 1}}
      />
    )).toMatchSnapshot();
  });
});
