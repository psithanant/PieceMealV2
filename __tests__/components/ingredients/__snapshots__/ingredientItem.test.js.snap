// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the IngredientItem component renders IngredientItem component properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#ccc",
      "borderRadius": 2,
      "borderWidth": 0.5,
      "elevation": 3,
      "flex": 1,
      "flexWrap": "wrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
  virtual={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "width": 56,
          }
        }
        virtual={undefined}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#373737",
            "fontFamily": "Futura",
            "fontSize": 16,
            "marginLeft": 25,
          }
        }
        virtual={undefined}
      >
        
      </Text>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 0.8,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "alignSelf": null,
                "color": "#68BAA7",
                "fontSize": 22,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          virtual={undefined}
        >
          Ôèë
        </Text>
      </View>
    </View>
  </View>
</View>
`;
